### PR TITLE
Only support keys as stdin and move tests

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#1603 

# Overview

- [x] **Breaking**: Only accept key inputs as stdin for `key inspect`, `key public`, `key child` (no arguments)
- [x] Moved unit parser tests to be full integration tests
- [x] Tweaked some help text

# Comments

### CODE REVIEW

File: lib/cli/test/unit/Cardano/CLISpec.hs [hunk 1]
```diff
@@ -227,8 +220,18 @@ spec = do
             , "  root                     Extract root extended private key from"
             , "                           a mnemonic sentence."
             , "  child                    Derive child keys."
-            , "  public                   Extract public key from a private key."
+            , "  public                   Extract the public key from a private"
+            , "                           key."
             , "  inspect                  Show information about a key."
+            , ""
+            , "Keys are read from standard input for convenient chaining of commands."
```
Comment (Anviking) [inline lib/cli/test/unit/Cardano/CLISpec.hs:227]: Thought it looked weird above the `Usage`, so I moved it to the footer.

Comment (rvl) [reply lib/cli/test/unit/Cardano/CLISpec.hs:227]: Yep, this is better.

**Review by rvl — APPROVED** [review]
:+1:

**Review by KtorZ — APPROVED** [review]

**Pull request merged**